pkg/types/conversion/installconfig.go

```diff
@@ -73,9 +73,15 @@ func ConvertNetworking(config *types.InstallConfig) {
 
 // ConvertBaremetal upconverts deprecated fields in the baremetal
 // platform. ProvisioningDHCPExternal has been replaced by setting
-// the ProvisioningNetwork field to "Unmanaged"
+// the ProvisioningNetwork field to "Unmanaged" and ProvisioningHostIP
+// has been replaced by ClusterProvisioningIP.
 func ConvertBaremetal(config *types.InstallConfig) {
 	if config.Platform.BareMetal.DeprecatedProvisioningDHCPExternal == true && config.Platform.BareMetal.ProvisioningNetwork == "" {
 		config.Platform.BareMetal.ProvisioningNetwork = baremetal.UnmanagedProvisioningNetwork
 	}
+
+	if config.Platform.BareMetal.DeprecatedProvisioningHostIP != "" && config.Platform.BareMetal.ClusterProvisioningIP == "" {
+		config.Platform.BareMetal.ClusterProvisioningIP = config.Platform.BareMetal.DeprecatedProvisioningHostIP
+		config.Platform.BareMetal.DeprecatedProvisioningHostIP = ""
```

Comment:
any reason why we would reset the deprecated field when that was set by the user?

Reply:
instead, i think we should add a validation that checks when both are set by the user they are equal.

Reply:
Is there a precedent for removing a field? Or will the Deprecated parameters always stay?

It makes sense to me to copy the value to the new field, reset to the zero-value and make sure it is omitempty, so it doesn't show up in the install-config any more.  Users don't have to worry about remembering which is the deprecated one.

It would probably be wortwhile formalizing the deprecation/removal procedure in the dev docs so we can all do it consistently.